src/overlays/actors/ovl_Magic_Wind/z_magic_wind.c

```diff
@@ -15,39 +15,244 @@ void MagicWind_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void MagicWind_Update(Actor* thisx, GlobalContext* globalCtx);
 void MagicWind_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
 const ActorInit Magic_Wind_InitVars = {
-    ACTOR_MAGIC_WIND,
-    ACTORTYPE_ITEMACTION,
-    FLAGS,
-    OBJECT_GAMEPLAY_KEEP,
-    sizeof(MagicWind),
-    (ActorFunc)MagicWind_Init,
-    (ActorFunc)MagicWind_Destroy,
-    (ActorFunc)MagicWind_Update,
-    (ActorFunc)MagicWind_Draw,
+    ACTOR_MAGIC_WIND,  ACTORTYPE_ITEMACTION, FLAGS,          OBJECT_GAMEPLAY_KEEP, sizeof(MagicWind), MagicWind_Init,
+    MagicWind_Destroy, MagicWind_Update,     MagicWind_Draw,
+};
+
+#include "z_magic_wind_gfx.c"
+
+u8 sTransformRefIdx[] = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+s16 sCopyValues[] = {
+    0x0400, 0x0400, 0x0400, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0200, 0x0200, 0x0000, 0x0000,
+    0x0000, 0x0000, 0x0000, 0x0000, 0x02CD, 0x02CD, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000,
+};
+
+TransformData sTransformData[] = {
+    {
+        0x000C,
+        0x0001,
+        0x0001,
+        0x0001,
+        0.0f,
+    },
+    {
+        0x0014,
+        0x003C,
+        0x0000,
+        0x0000,
+        1.5f,
+    },
+    {
+        0x000C,
+        0x0001,
+        0x0001,
+        0x0001,
+        0.0f,
+    },
+    {
+        0x0014,
+        0x003C,
+        0x0000,
+        0x0000,
+        1.0f,
+    },
+};
+
+TransformUpdateIndex sTransformUpdIdx = {
+    sTransformRefIdx, sTransformData, sCopyValues, 0x0001, 0x0003C,
+};
+
+SkelCurveLimb sRootLimb = {
+    0x01,
+    0xFF,
+    {
+        NULL,
+        NULL,
+    },
+};
+
+SkelCurveLimb sInnerCylinder = {
+    0xFF,
+    0x02,
+    {
+        NULL,
+        sInnerCylinderDList,
+    },
+};
+
+SkelCurveLimb sOuterCylinder = {
+    0xFF,
+    0xFF,
+    {
+        NULL,
+        sOuterCylinderDList,
+    },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Magic_Wind/func_80B8B100.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Magic_Wind/MagicWind_Init.s")
+SkelCurveLimb* sLimbs[] = {
+    &sRootLimb,
+    &sInnerCylinder,
+    &sOuterCylinder,
+};
+
+SkelCurveLimbList sLimbList = {
+    sLimbs,
+    0x03,
+};
+
+u8 sAlphaUpdVals[] = {
+    0x00, 0x03, 0x04, 0x07, 0x09, 0x0A, 0x0D, 0x0F, 0x11, 0x12, 0x15, 0x16, 0x19, 0x1B, 0x1C, 0x1F, 0x21, 0x23,
+};
+
+void MagicWind_Shrink(MagicWind* this, GlobalContext* globalCtx);
+void MagicWind_WaitForTimer(MagicWind* this, GlobalContext* globalCtx);
+void MagicWind_FadeOut(MagicWind* this, GlobalContext* globalCtx);
+void MagicWind_FullSize(MagicWind* this, GlobalContext* globalCtx);
+void MagicWind_Grow(MagicWind* this, GlobalContext* globalCtx);
```

Comment:
We normally have all necessary function prototypes at the top with the rest of them.